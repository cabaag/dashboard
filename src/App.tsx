import { ThemeProvider } from '@material-ui/core';
import React from 'react';
import {
	BrowserRouter as Router,
	Switch
} from 'react-router-dom';
import './App.scss';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import { PublicRoute } from './components/PublicRoute/PublicRoute';
import theme from './config/theme';
import { GlobalContext } from './globalContext';
import Home from './pages/Home/Home';
import Login from './pages/Login/Login';

const tokenTmp = localStorage.getItem('token') || null;

function App() {
	const [token, setToken] = React.useState<string | null>(tokenTmp);

	const handleSetToken = (t: string | null) => {
		setToken(t);
	};

	return (
		<GlobalContext.Provider value={{ token, setToken: handleSetToken }}>
			<ThemeProvider theme={theme}>
				<Router>
					<Switch>
						<PublicRoute exact path='/login' component={Login} />
						<PrivateRoute path='/' component={Home} />
					</Switch>
				</Router>
			</ThemeProvider>
		</GlobalContext.Provider>
	);
}

export default App;
