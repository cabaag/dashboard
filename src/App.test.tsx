import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

describe('test App component', () => {
	test('should match snapshot', () => {
		expect(shallow(<App />)).toMatchSnapshot();
	});

});
