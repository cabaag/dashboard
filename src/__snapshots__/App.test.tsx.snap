// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App component should match snapshot 1`] = `ShallowWrapper {}`;
